core/src/test/scala/pureconfig/PureconfSuite.scala

```diff
@@ -413,6 +413,29 @@ class PureconfSuite extends FlatSpec with Matchers with OptionValues with TryVal
     loadConfig[ConfWithListOfFoo](conf) shouldBe Success(expected)
   }
 
+  it should "be able to load a set of Ints from the system properties" in {
+    System.setProperty("pure.conf.intSet.0", "1")
```

Comment:
Please use one of the other methods for loading configuration that doesn't modify global state, such as `ConfigFactory.parseString`. While I recognize that you made an effort to clear those settings, a failed test assertion would throw an exception, which would prevent the global state from being clear. Furthermore it's simply brittle, requiring a dev to make changes two places when modifying a test.